internal/allocator/allocator.go

```diff
@@ -54,6 +54,60 @@ type alloc struct {
 	key
 }
 
+// ipPool represents a mapping of IP families (IPv4, IPv6, or both) to their respective IP addresses.
+// This type is used to select and organize IPs based on a service's IP family policy.
+// Depending on the family policy, the map may contain:
+// - Just an IPv4 address
+// - Just an IPv6 address
+// - Both IPv4 and IPv6 addresses for dual-stack support.
+//
+// Key:
+// - ipfamily.IPv4: Corresponds to the IPv4 address allocated for the service.
+// - ipfamily.IPv6: Corresponds to the IPv6 address allocated for the service.
+type ipPool struct {
+	PoolName string
+	IPMap    map[ipfamily.Family]net.IP
```

Comment:
if this is (at most) a 2 keys map, I'd suggest to have to explicit fields and a get / set method with ip and  family as parameters. It will make it easier to understand what this object is about.